<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>An app holding the permission <code>REQUEST_IGNORE_BATTERY_OPTIMIZATIONS</code> asks the user to allow it to ignore
    battery optimizations (that is, put them on the whitelist of apps).</p>
<p>Most applications should not use this; there are many facilities provided by the platform for applications to
    operate correctly in the various power saving modes.</p>

<h2>Noncompliant Code Example</h2>
<pre>
    &lt;manifest&gt;
        &lt;uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" /&gt;
    &lt;/manifest&gt;
</pre>
