<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>A negative effect on the device s battery is when an app is paired with a companion device (over Bluetooth, BLE, or Wi-Fi)
    and that it has been excluded from battery optimizations (run in the background) using the declaration
    <code>Manifest.permission.REQUEST_COMPANION_RUN_IN_BACKGROUND</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    &lt;manifest&gt;
       &lt;uses-permission android:name="android.permission.REQUEST_COMPANION_RUN_IN_BACKGROUND"&gt;
    &lt;/manifest&gt;
</pre>