<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Services are long-living operations, as components of the apps. However, they can be started in isolation each time the device is next started,
    without the user's acknowledgement.<br>
    This technique should be discouraged because the accumulation of these silent services results in excessive battery depletion that remains
    unexplained from the end-user's point of view.<br>
    In addition, end-users know how to kill applications, but more rarely how to kill services.<br>
    Thus, any developer should avoid having a call to <code>Context#startService()</code> from a Broadcast Receiver component that has specified an
    intent-filter for the <code>BOOT_COMPLETED</code> action in the manifest.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    &lt;receiver&gt;
        &lt;intent-filter&gt;
            &lt;action android:name= "android.intent.action.BOOT_COMPLETED"/&gt;
        &lt;/intent-filter&gt;
    &lt;/receiver&gt;
</pre>