<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Developers are allowed to apply native themes for their app, or derive new ones through inheritance. This decision
    has a significant impact on energy consumption since displaying dark colors is particularly beneficial for mobile
    devices with (AM)OLED screens.<br>
    By default, Android will set Holo to the Dark theme (parent style <code>Theme.Holo</code>) and hence switching to the
    light theme (parent style <code>Theme.Holo.Light</code>) within the manifest or in xml themes should be avoided.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    &lt;style name="Theme.MyTheme" parent="@android:style/Theme.Holo.Light"/&gt;
</pre>
or
<pre>
    &lt;application android:label="@string/app_name" android:theme="@android:style/Theme.Holo.Light"/&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
    &lt;style name="Theme.MyTheme" parent="@android:style/Theme.Holo"/&gt;
</pre>
or
<pre>
    &lt;application android:label="@string/app_name" android:theme="@android:style/Theme.Holo"/&gt;
</pre>
