<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Creation of a Media Player object with <code>new MediaPlayer()</code> can
    be used to control playback of audio/video files and streams. Class own a<code>release()</code> method.</p>
<p>In addition to unnecessary resources (such as memory and instances of codecs) being held,
    failure to call this method immediately if a media object is no longer needed may also lead to
    continuous battery consumption for mobile devices.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    MediaPlayer mp = new MediaPlayer();
</pre>
<h2>Compliant Solution</h2>
<pre>
    MediaPlayer mp = new MediaPlayer();
    mp.release();
</pre>
