<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>
    Importing the <code>com.google.android.gms.analytics.Tracker</code> or <code>com.google.firebase.analytics.FirebaseAnalytics</code> classes means that the app sends hits to Google Analytics. It is not necessarily sensitive information, but it is a first step towards Google Ads and hence this practice should be discouraged at early stage.
</p>
<h2>Noncompliant Code Example</h2>
Import of Google Analytics :
<pre>
    import com.google.android.gms.analytics.Tracker
</pre>
Import of Firebase Analytics :
<pre>
    import com.google.firebase.analytics
</pre>