<img src="http://www.neomades.com/extern/partage/ecoCode/1sur5_1x.png">
<p>Opening and closing internet connection continuously is extremely battery-inefficient since HTTP exchange is the most
    consuming operation of the network. This bug typically occurs when one obtain a new <code>HttpURLConnection</code>
    by calling  <code>URL#openConnection()</code>  within a loop control structure (while, for, do-while, for-each).<br>
    Also, this bad practice must be early prevented because it is the root of another evil that consists in polling
    data at regular intervals, instead of using push notifications to save a lot of battery power.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    for (int val : myArray) {
        URL.openConnection();
    }
</pre>
