<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Shaking of an Android device is possible in all circumstances with a call to <code>getSystemService(Context.VIBRATOR_SERVICE)</code> (API 26) or
    <code>getSystemService(Context.VIBRATOR.MANAGER.SERVICE)</code> (API 31).
    Behind this effect stands a specific miniature hardware component, motor or actuator, that consumes power.
    As a consequence, its usage must be discouraged, especially since its added value is not clear.</p>
<h2>Noncompliant Code Example</h2>
API 26:
<pre>
    Vibrator v = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);
    v.vibrate(400);
</pre>
API 31:
<pre>
    VibratorManager vm = (VibratorManager) getSystemService(Context.VIBRATOR_MANAGER_SERVICE);
    vm.getDefaultVibrator().vibrate(VibrationEffect.createOneShot(500, VibrationEffect.DEFAULT_AMPLITUDE));
</pre>