<img src="http://www.neomades.com/extern/partage/ecoCode/3sur5_1x.png">
<p>The fused location provider is one of the location APIs in Google Play
    services which combines signals from GPS, Wi-Fi, and cell networks, as well
    as accelerometer, gyroscope, magnetometer and other sensors. It is officially
    recommended to maximize battery life.<br>
    Thus, developer has to set up Google Play Service in her gradle file with a dependency to
    <code>com.google.android.gms:play-services-location:x.y.z</code>, and then to
    import from <code>com.google.android.gms.location</code> instead of the
    <code>android.location</code> package of the SDK.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    import android.location.*;
</pre>
<h2>Compliant Solution</h2>
<pre>
    import com.google.android.gms.location.*;
</pre>