<img src="http://www.neomades.com/extern/partage/ecoCode/3sur5_1x.png">
<p>Normally the Wi-Fi stack filters out packets not explicitly addressed to the device. Acquiring a Multicast Lock with
    <code>WifiManager.MulticastLock.acquire()</code> will cause the stack to receive packets addressed to multicast
    addresses.</p>
<p>Processing these extra packets can cause a noticeable battery drain and must be disabled when not needed
    with to a call to <code>WifiManager.MulticastLock.release()</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    WifiManager wifiManager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
    WifiManager.MulticastLock lock = wifiManager.createMulticastLock("tag");
    lock.acquire();
</pre>
<h2>Compliant Code Example</h2>
<pre>
    WifiManager wifiManager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
    WifiManager.MulticastLock lock = wifiManager.createMulticastLock("tag");
    lock.acquire();
    lock.release()
</pre>
