<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Introduced in Android 9, the adaptive brightness feature raises or lower the brightness of the screen depending on
    the light in the current environment.<br>
    For some reasons, developers may disable this feature programmatically, setting the field
    <code>WindowManager.LayoutParams#screenBrightness</code> with the constant <code>BRIGHTNESS_OVERRIDE_FULL</code>.<br>
    This feature was introduced to improve battery life, be careful when deactivating it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    getWindow().getAttributes().screenBrightness = BRIGHTNESS_OVERRIDE_FULL;
</pre>
