<img src="http://www.neomades.com/extern/partage/ecoCode/1sur5_1x.png">
<p>Most Android-powered devices have built-in sensors that measure motion, orientation, and various environmental conditions.
    In addition to these are the image sensor (a.k.a. Camera) and the geo-positioning sensor (a.k.a. GPS).</p>
<p>The common point of all these sensors is that they are expensive while in use. Their common bug is to let the sensor
    unnecessarily process data when the app enters an idle state, typically when paused or stopped.<br>
    Consequently, calls must be carefully pairwised: <code>SensorManager#registerListener()/unregisterListener()</code>.
    Failing to do so can drain the battery in just a few hours.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    SensorManager sManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
    Sensor accelerometer = sManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
    sManager.registerListener(this,accelerometer,SensorManager.SENSOR_DELAY_NORMAL);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    SensorManager sManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
    Sensor accelerometer = sManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
    sManager.registerListener(this,accelerometer,SensorManager.SENSOR_DELAY_NORMAL);
    sManager.unregisterListener(this);
</pre>

