<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>In Android 11 (API level 30) or higher, a call to
    <code>Surface#setFrameRate(float frameRate, int compatibility)</code> results in a change to the display refresh
    rate. However, a regular app displays 60 frames per second (60Hz). In order to optimize content refreshes and hence
    saving energy, this frequency should not be raised to 90Hz or 120Hz, despite this is now supported by many devices.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    surface.setFrameRate(120f, FRAME_RATE_COMPATIBILITY_DEFAULT);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    surface.setFrameRate(60f, FRAME_RATE_COMPATIBILITY_DEFAULT);
</pre>