<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Creation of a Media Recorder object with <code>new MediaRecorder()</code> is used to record
    audio and video. Class own a <code>release()</code> method.</p>
<p>In addition to unnecessary resources (such as memory and instances of codecs) being held,
    failure to call this method immediately if a media object is no longer needed may also lead to
    continuous battery consumption for mobile devices.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    MediaRecorder mr = new MediaRecorder();
</pre>
<h2>Compliant Solution</h2>
<pre>
    MediaRecorder mr = new MediaRecorder();
    mr.release();
</pre>
