<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Turning on the torch mode programmatically with <code>CameraManager#setTorchMode(..., true)</code> must absolutely be
    avoided because the flashlight is one of the most energy-intensive component.</p>
<h2>Noncompliant Code Example</h2>
<pre>
 CameraManager camManager = (CameraManager) getSystemService(Context.CAMERA_SERVICE);
        String cameraId = null;
        try {
            cameraId = camManager.getCameraIdList()[0];
            camManager.setTorchMode(cameraId, true);
        } catch (CameraAccessException e) {
            e.printStackTrace();
        }
</pre>