<img src="http://www.neomades.com/extern/partage/ecoCode/3sur5_1x.png">
<p>To avoid draining the battery, an Android device that is left idle quickly falls asleep.<br>
    Hence, keeping the screen on should be avoided, unless it is absolutely necessary. If so, developers typically
    use the <code>FLAG_KEEP_SCREEN_ON</code> in their activity. Another way to implement this is in their application's
    layout XML file, by using the <code>android:keepScreenOn</code> attribute.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    getWindow().setFlags(FLAG_KEEP_SCREEN_ON, FLAG_KEEP_SCREEN_ON);
</pre>
