<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>To avoid draining the battery, an Android device that is left idle quickly falls asleep.<br>
    Hence, keeping the screen on should be avoided, unless it is absolutely necessary. If so, developers typically
    use the <code>FLAG_KEEP_SCREEN_ON</code> in their activity. Another way to implement this is in their application's
    layout XML file, by using the <code>android:keepScreenOn</code> attribute.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    &lt;RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:keepScreenOn="true"/&gt;
</pre>
<h2>Compliant Code Example</h2>
<pre>
    &lt;RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:keepScreenOn="false"/&gt;
</pre>
