<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Displaying dark colors is particularly beneficial for mobile devices with (AM)OLED screens.<br>
    Consequently, custom resources like bright colors values and bitmap images with too high luminance should be avoided.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    &lt;TextView
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:foreground="#FFFFFF"
        android:foregroundTint="#FFFFFF"&gt;
    &lt;/TextView&gt;
</pre>