<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Turning on the torch mode programmatically must absolutely be avoided because the flashlight is one of the most energy-intensive component.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    AVCaptureTorchMode.on
</pre>
<pre>
    "setTorchModeOn(level: Float)"</pre>
<pre>
    "torchMode = .on"
</pre>