<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Allowing the system to pause location updates, at times when the location data is unlikely to change, can improve
    battery life without sacrificing location data. This is the default behavior. For example, if the user stops for
    food while using a navigation app, the location manager might pause updates for a period of time. It is possible
    to disable this behavior, by setting the <code>CLLocationManager#pausesLocationUpdatesAutomatically</code>
    property to <code>false</code>, but applications are strongly discouraged of doing it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    let manager = CLLocationManager()
    manager.pausesLocationUpdatesAutomatically = false
</pre>

