<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>
    For some use cases, it might be necessary to get a unique device identifier by a call to <code>TelephonyManager#getDeviceId()</code>
    (returns IMEI on GSM, MEID for CDMA).
    However, this raises privacy concerns and it is not recommended.
    Alternatively, you may use android.provider.Settings.Secure.ANDROID_ID.
</p>
<h2>Noncompliant Code Example</h2>
Use of: TelephonyManager#.getDeviceId()
<pre>
    TelephonyManager telephonyManager = (TelephonyManager) getSystemService(Context.TELEPHONY_SERVICE);
    String deviceId = telephonyManager.getDeviceId()
</pre>
Use :
<pre>
    android.provider.Settings.Secure.ANDROID_ID
</pre>

