<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>With a call to
    <code>LocationManager#requestLocationUpdates(String provider, long minTime, float minDistance, LocationListener listener)</code>,
    the location provider will only send your application an update when the location has changed by at
    least minDistance meters AND at least minTime milliseconds have passed. So <code>minTime</code> should
    be the primary tool to conserve battery life, and, to a lesser extent, <code>minDistance</code>.<br>
    These two must imperatively be greater than 0.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER,
                0,
                0,
                this);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER,
                60000L,  // refresh location at least each 60000ms
                10.0f,   // refresh location at least each 10 meters
                this);
</pre>