<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Giving information to the end-user through notifications is an important aspect of a modern app.<br>
    However, a notification does not necessarily need to be loud and vibrant to achieve its purpose.
    Default mode is enough. That is why when building a notification with
    <code>NotificationManager</code> there should be no extra calls to the methods <code>setSound()</code> nor
    <code>setVibrate()</code> (API26) nor <code>setVibrationPattern()</code> (API31) on the builder object.</p>
<h2>Noncompliant Code Example</h2>
API 26:
<pre>
    Notification.Builder notificationBuilder = new Notification.Builder(getApplicationContext(), "42");
    notificationBuilder.setVibrate(new long[] {1000, 1000, 1000, 1000, 1000});
    notificationBuilder.setSound(
        RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION),
        Notification.AUDIO_ATTRIBUTES_DEFAULT
    );
</pre>
API 31:
<pre>
    NotificationChannel notification = new NotificationChannel("42",
        "test",
        NotificationManager.IMPORTANCE_DEFAULT
    );
    notification.setVibrationPattern(new long[]{1000, 1000, 1000, 1000, 1000});
    notification.setSound(
        RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION),
        Notification.AUDIO_ATTRIBUTES_DEFAULT
    );
</pre>
