<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>During a voice interaction session, <code>VoiceInteractionSession#setKeepAwake(boolean keepAwake)</code> allows to decide whether it will keep the
    device awake while it is running a voice activity.</p>
<p>By default, the system holds a wake lock for it while in this state, so that it can work even if the screen is off. Setting this to
    <code>false</code> removes that wake lock, allowing the CPU to go to sleep and hence does not let this continue to drain the battery.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    VoiceInteractionSession voiceSession = new VoiceInteractionSession(this);
</pre>
OR
<pre>
    VoiceInteractionSession voiceSession = new VoiceInteractionSession(this);
    voiceSession.setKeepAwake(true);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    VoiceInteractionSession voiceSession = new VoiceInteractionSession(this);
    voiceSession.setKeepAwake(false);
</pre>
