<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Applications are strongly discouraged from using exact alarms unnecessarily as they reduce the OS's
    ability to minimize battery use (i.e. Doze Mode).</p>
<p>For most apps prior to API 19, <code>setInexactRepeating()</code> is preferable over
    <code>setRepeating()</code>. When you use this method, Android synchronizes multiple inexact
    repeating alarms and fires them at the same time, thus reducing the battery drain.</p>
<p>Similarly after API 19, <code>setExact()</code> and <code>setExactAndAllowWhileIdle()</code>
    can significantly impact the power use of the device when idle, so they should be used with
    care. High-frequency alarms are also bad for battery life but this is already checked by Android
    lint (ShortAlarm built-in check).</p>
<h2>Noncompliant Code Example</h2>
<pre>
    AlarmManager alarmManager = (AlarmManager) this.getSystemService(Context.ALARM_SERVICE);
    alarmManager.setRepeating(alarmType, triggerAtMillis, intervalMillis, operation);
</pre>
or
<pre>
    AlarmManager alarmManager = (AlarmManager) this.getSystemService(Context.ALARM_SERVICE);
    alarmManager.setExact(type,triggerAtMillis,operation);
</pre>
or
<pre>
    AlarmManager alarmManager = (AlarmManager) this.getSystemService(Context.ALARM_SERVICE);
    alarmManager.setExact(type,triggerAtMillis,tag,listener,targetHandler);
</pre>
or
<pre>
    AlarmManager alarmManager = (AlarmManager) this.getSystemService(Context.ALARM_SERVICE);
    alarmManager.setExactAndAllowWhileIdle(type,triggerAtMilllis,operation);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    AlarmManager alarmManager = (AlarmManager) this.getSystemService(Context.ALARM_SERVICE);
    alarmManager.setInexactRepeating(alarmType, triggerAtMillis, intervalMillis, operation);
</pre>
