<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>With Bluetooth Low Energy technology (see BLE API smell), a Bluetooth Smart Ready
    device (the master) will establish a link with a Bluetooth Smart device (the slave).
    Most often, the slave is a GATT server and the master is a GATT client. GATT capable
    devices can be discovered using BLE scan process.</p>
<p>Invoking <code>BluetoothGatt#requestConnectionPriority(int connectionPriority)</code>
    with the value <code>CONNECTION_PRIORITY_LOW_POWER</code>, is recommended to reduce
    power consumption.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    BluetoothGatt gatt = new BluetoothGatt();
    gatt.requestConnectionPriority(CONNECTION_PRIORITY_HIGH);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    BluetoothGatt gatt = new BluetoothGatt();
    gatt.requestConnectionPriority(CONNECTION_PRIORITY_LOW_POWER);
</pre>