<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>The Android 5.0 Lollipop (API 21) release introduces a job scheduler API via the Job Scheduler class.<br>
    Compared to a custom Sync Adapter or the alarm manager, the Job Scheduler supports batch scheduling of jobs.<br>
    The Android system can combine jobs so that battery consumption is reduced.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    void setAlarm(Context context)
    {
        AlarmManager alarmManager = (AlarmManager)context.getSystemService(Context.ALARM_SERVICE);
        Intent intent = new Intent(context, Alarm.class);
        PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 0, intent, 0);
        alarmManager.setRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), 1000 * 60 * 10, pendingIntent);
    }

    Alarm alarm = new Alarm();
    alarm.setAlarm(this);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    JobInfo info = new JobInfo.Builder(123, COMPONENT_NAME)
                .setRequiresCharging(true)
                .setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED)
                .setPersisted(true)
                .setPeriodic(10 * 60 * 1000)
                .build();
    JobScheduler scheduler = (JobScheduler) getSystemService(JOB_SCHEDULER_SERVICE);
    scheduler.schedule(info);
</pre>
