<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Transmitting a file over a network infrastructure without compressing
    it consumes more energy than with compression. More precisely,
    energy efficiency is improved in case the data is compressed at least by 10%,
    transmitted and decompressed at the other network node. From the Android client side,
    it means making a post HTTP request using a <code>GZIPOutputStream</code>
    instead of the classical OutputStream, along with the <code>HttpURLConnection</code> object.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    Url url = new Url("https://www.ecocode.io/");
    HttpsUrlConnection con = (HttpsURLConnection) url.openConnection();
    OutputStream stream = con.getOutputStream();
</pre>
<h2>Compliant Code Example</h2>
<pre>
    Url url = new Url("https://www.ecocode.io/");
    HttpsUrlConnection con = (HttpsURLConnection) url.openConnection();
    OutputStream stream = new GZIPOutputStream(con.getOutputStream());
</pre>
