<img src="http://www.neomades.com/extern/partage/ecoCode/3sur5_1x.png">
<p>In contrast to classic Bluetooth, Bluetooth Low Energy (BLE)
    is designed to provide significantly lower power consumption.
    Its purpose is to save energy on both paired devices but very few
    developers are aware of this alternative API.
    From the Android client side, it means append <code>android.bluetooth.le.*</code>
    imports to <code>android.bluetooth.*</code> imports in order to benefits from
    low-energy features.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    import android.bluetooth.*
</pre>
<h2>Compliant Solution</h2>
<pre>
    import android.bluetooth.le.*
</pre>