<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>For developers wishing to display OpenGL rendering, when choosing the rendering mode
    with <code>GLSurfaceView#setRenderMode(int renderMode)</code>,
    using <code>RENDERMODE_WHEN_DIRTY</code> instead of <code>RENDERMODE_CONTINUOUSLY</code>
    (by default) can improve battery life and overall system performance by allowing the GPU and CPU
    to idle when the view does not need to be updated.</p>
<h2>Noncompliant Code Example</h2>
<pre>
     GLSurfaceView surfaceView = new GLSurfaceView(this);
     surfaceView.setRenderMode(GLSurfaceView.RENDERMODE_CONTINUOUSLY)
</pre>
<br>
<h2>Compliant Code Example</h2>
<pre>
     GLSurfaceView surfaceView = new GLSurfaceView(this);
     surfaceView.setRenderMode(GLSurfaceView.RENDERMODE_WHEN_DIRTY)
</pre>
