<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>With Bluetooth Low Energy technology, a Bluetooth Smart Ready device (the master)
    will establish a link with a Bluetooth Smart device (the slave). Most often,
    the slave is a GATT server and the master is a GATT client. GATT capable
    devices can be discovered using BLE scan process.</p>
<p>The default and preferred advertising mode is <code>ADVERTISE_MODE_LOW_POWER</code>
    when calling <code>AdvertiseSettings.Builder#setAdvertiseMode(int advertiseMode).</code>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    AdvertiseSettings.Builder builder = new AdvertiseSettings.Builder();
    builder.setAdvertiseMode(AdvertiseSettings.ADVERTISE_MODE_LOW_LATENCY);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    AdvertiseSettings.Builder builder = new AdvertiseSettings.Builder();
    builder.setAdvertiseMode(AdvertiseSettings.ADVERTISE_MODE_LOW_POWER);
</pre>