<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>The rotation vector sensor is the most frequently used sensor for motion detection and monitoring. When using
    <code>SensorManager#getDefaultSensor(int type)</code>, always prefer the
    constant <code>TYPE_GEOMAGNETIC_ROTATION_VECTOR</code> which is similar to <code>TYPE_ROTATION_VECTOR</code>,
    but using a magnetometer instead of using a gyroscope. This sensor uses lower power than the other rotation vectors,
    because it doesn't use the gyroscope. However, it is more noisy and will work best outdoors.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    SensorManager sManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
    sManager.getDefaultSensor(Sensor.TYPE_ROTATION_VECTOR);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    SensorManager sManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
    sManager.getDefaultSensor(Sensor.TYPE_GEOMAGNETIC_ROTATION_VECTOR);
</pre>
